t/basic-consumer.t

```diff
@@ -0,0 +1,268 @@
+        ';
+    }
+--- request
+GET /t
+--- response_body
+test-consumer: partition 0, offset: 0LL
+test-consumer: partition 1, offset: 0LL
+--- no_error_log
+[error]
+
+
+
+=== TEST 3: list offset (last)
+--- http_config eval: $::HttpConfig
+--- config
+    location /t {
+        content_by_lua '
+            ngx.sleep(1) -- wait 1 second for kafka
+            local cjson = require("cjson")
+            local bconsumer = require("resty.kafka.basic-consumer")
+            local protocol_consumer = require("resty.kafka.protocol.consumer")
+
+            local broker_list = {
+                { host = "$TEST_NGINX_KAFKA_HOST", port = $TEST_NGINX_KAFKA_PORT },
+            }
+
+            local c = bconsumer:new(broker_list)
+
+            local offset0, err = c:list_offset("test-consumer", 0, protocol_consumer.LIST_OFFSET_TIMESTAMP_LAST)
+            if not offset0 then
+                ngx.say(err)
+                return
+            end
+
+            ngx.say("test-consumer: partition 0, offset: ", offset0.offset)
+
+            local offset1, err = c:list_offset("test-consumer", 1, protocol_consumer.LIST_OFFSET_TIMESTAMP_LAST)
+            if not offset1 then
+                ngx.say(err)
+                return
+            end
+
+            ngx.say("test-consumer: partition 1, offset: ", offset1.offset)
+        ';
+    }
+--- request
+GET /t
+--- response_body
+test-consumer: partition 0, offset: 67LL
+test-consumer: partition 1, offset: 68LL
+--- no_error_log
+[error]
+
+
+
+=== TEST 4: fetch message (first)
+--- http_config eval: $::HttpConfig
+--- config
+    location /t {
+        content_by_lua '
+            local cjson = require("cjson")
+            local bconsumer = require("resty.kafka.basic-consumer")
+            local protocol_consumer = require("resty.kafka.protocol.consumer")
+
+            local broker_list = {
+                { host = "$TEST_NGINX_KAFKA_HOST", port = $TEST_NGINX_KAFKA_PORT },
+            }
+
+            local c = bconsumer:new(broker_list)
+
+            local ret0, err = c:fetch("test-consumer", 0, 0) -- partition 0, offset 0
+            local message0 = ""
+            for _, record in pairs(ret0.records) do
+                message0 = message0 .. record.value
+            end
+            ngx.say(message0)
+
+            local ret1, err = c:fetch("test-consumer", 1, 0) -- partition 1, offset 0
+            local message1 = ""
+            for _, record in pairs(ret1.records) do
+                message1 = message1 .. record.value
+            end
+            ngx.say(message1)
+        ';
+    }
+--- request
+GET /t
+--- response_body
+msg2msg4msg6msg8msg10msg12msg14msg16msg18msg20msg22msg24msg26msg28msg30msg32msg34msg36msg38msg40msg42msg44msg46msg48msg50msg52msg54msg56msg58msg60msg62msg64msg66msg68msg70msg72msg74msg76msg78msg80msg82msg84msg86msg88msg90msg92msg94msg96msg98msg100msg102msg104msg106msg108msg110msg112msg114msg116msg118msg120msg122msg124msg126msg128msg130msg132msg134
```

Comment:
it is long. can we use a smaller case?

Reply:
I removed the prefix `msg` and now the checksum length is much lower.